Sources/XCTest/Private/ArgumentParser.swift

```diff
@@ -52,21 +52,31 @@ internal struct ArgumentParser {
 
     init(arguments: [String]) {
         self.arguments = arguments
-    }
-
-    var executionMode: ExecutionMode {
-        if arguments.count <= 1 {
-            return .run(selectedTestNames: nil)
-        } else if arguments[1] == "--list-tests" || arguments[1] == "-l" {
-            return .list(type: .humanReadable)
-        } else if arguments[1] == "--dump-tests-json" {
-            return .list(type: .json)
-        } else if arguments[1] == "--help" || arguments[1] == "-h" {
-            return .help(invalidOption: nil)
-        } else if let fst = arguments[1].first, fst == "-" {
-            return .help(invalidOption: arguments[1])
-        } else {
-            return .run(selectedTestNames: arguments[1].split(separator: ",").map(String.init))
+        for var i in arguments.dropFirst().indices {
+            let argument = arguments[i]
+            switch argument {
+            case "--list-tests", "-l":
+                executionMode = .list(type: .humanReadable)
+            case "--dump-tests-json":
+                executionMode = .list(type: .json)
+            case "--help", "-h":
+                executionMode = .help(invalidOption: nil)
+            case "--testing-library":
+                // Ignore this argument, it's already been handled by the main() that called into XCTMain().
+                i = arguments.index(after: i) // skip value
```

Comment:
Is this intended to "consume" the following argument in the list after `--testing-library`? Apologies if I'm misunderstanding something but won't the next turn of the `for` loop reassign `i` to the next index anyway?